pkg/util/updater/updater.go

```diff
@@ -86,60 +88,64 @@ const (
 	}
 	defer atomic.StoreInt32(&u.updating, 0)
 
-	u.status.Set("Started, checking update")
-
-	version, err := u.getVersion(updateConfig)
+	current, err := user.Current()
 	if err != nil {
-		return false, err
-	}
-
-	// No update is available.
-	if version == "" {
-		return false, nil
+		u.status.Set("failed checking permissions")
+		u.log.Error("failed checking permissions")
+		return false, errors.New("failed checking permissions")
 	}
-
-	u.status.Set(fmt.Sprintf("Found version %q", version))
-
-	u.status.Set(fmt.Sprintf("Checking/Adding repo %s", "https://deb.skywire.skycoin.com/")) // add if not exist
-	if err := u.addRepo(); err != nil {
-		return false, err
+	if current.Uid != "0" {
+		u.status.Set("insufficient permissions")
+		u.log.Error("insufficient permissions")
+		return false, errors.New("insufficient permissions")
 	}
 
-	u.status.Set("Update repositories")
-	if err := u.aptUpdate(); err != nil {
-		return false, err
+	var si sysinfo.SysInfo
+	si.GetSysInfo()
+	if (si.OS.Vendor != "debian") && (si.OS.Vendor != "arch") {
+		u.status.Set(fmt.Sprintf("updates not supported on this operating system: %s", si.OS.Vendor))
+		return false, errors.New("operating system not supported")
 	}
-	u.log.Info("Updating repositories by 'apt update' compeleted.")
 
-	// uninstall current installed skywire if its version is equal or lower that 0.4.2
-	currentVersion, err := currentVersion()
-	if err != nil {
-		return false, err
-	}
-	if currentVersion.Minor == 4 && currentVersion.Patch <= 2 {
-		u.status.Set("Uninstall current skywire version") // if needed
-		if err := u.aptRemove(); err != nil {
+	if si.OS.Vendor == "debian" {
+		u.status.Set("Checking for available package")
+		available, err := script.Exec(`apt-cache search skywire-bin`).String()
+		if err != nil {
+			return false, err
+		}
+		if (available == "") || (available == "\n") {
+			return false, errors.New("Repository not configured or skywire not available")
+		}
+		u.status.Set("syncing package database")
+		if _, err := script.Exec(`sudo apt update`).String(); err != nil {
+			u.log.Error("error syncing package database")
+			return false, err
+		}
+		u.log.Info("synced package database")
+		u.log.Info("installing skywire-bin")
+		if _, err := script.Exec(`sudo NOAUTOCONFIG=true apt install skywire-bin -y`).String(); err != nil {
+			u.log.Error("error installing skywire-bin")
+			return false, err
+		}
+		u.log.Info("installed skywire-bin")
+		u.log.Info("updating config and restarting visor")
+		if _, err := script.Exec(`DMSGPTYTERM=true skywire-autoconfig`).String(); err != nil {
+			u.log.Error("error running skywire-autoconfig")
 			return false, err
 		}
-		u.log.Info("Uninstalling old version compeleted.")
+		//this may not return before the process is restarted
+		return true, nil
 	}
-
-	u.status.Set(fmt.Sprintf("Installing Skywire %q", version))
-	if err := u.aptInstall(); err != nil {
-		return false, err
+	if si.OS.Vendor == "arch" {
```

Comment:
We are not going to implicitly support arch here in the binaries. Please take this out.

Reply:
This i will need to implement for testing. None of the code i added for updating .deb was tested yet because i am writing it on arch.

I can remove it before this gets merged or we can discuss more of an appropriate context for this, but if i am able to implement this, it would let me much more easily test the updater on the system it is being developed on.

all of the .deb packages, everything i have done, was first done on arch and tested there and then re-implemented for debian.

it may not be able to be implemented, because of how permissions need to be handled for invoking `yay` on archlinux, but i need to try.